articles/ros2_sectest.md

```diff
@@ -0,0 +1,188 @@
+
+## Vulnerability Evaluation Process
+
+Assessing and mitigating a vulnerability is a 4-step process. Its output is an attack proof-of-concept, a mitigation and a report (see appendix).
+
+1. **Attack PoC** - a proof of concept for the attack is implemented and contributed back to a Git repository listing all potential attacks.
+2. **Attack Validation** - the attack implemented in 1. is used to disrupt, take control or attack in any way a robot. The target platform is unimportant. For commodity reasons, we suggest to use TurtleBot3 as a reference platform for this step. Note: Not all attacks require validation on a robot. A conscious effort should be made during this step to avoid over-relying on any particular robot specific characteristic. If this is the case, those assumptions should be made explicit in the vulnerability report. This attack validation needs to be easily reproducible by the community, but does not need to be automated (i.e. it is OK to ask the user to check manually whether the robot is moving differently due to the attack for instance).
+3. **Mitigation PoC** - the mitigation, from a practical standpoint, means an improvement (PR) contributed back to any ROS 2 package.
+4. **Mitigation Validation** - Once the PR in 3. is merged, the step 2. should be repeated. At this point, the attack code should **not** be able to disrupt or impact the robot in any way. if the mitigation is partial (requires particular setup or configuration or does not cover all cases), this should be explicitly called out in the vulnerability report.
+
+## Vulnerability Evaluation Process Scope
+
+### ROS 2 specific VS Generic Vulnerabilities
+
+Unlike the ALIAS Robotic Vulnerability Disclosure Program [RVDP][RVDP], this security assessment program focuses on vulnerabilities *directly related to ROS 2*.
+
+For instance, the following issues are in scope:
+
+* Leaking ROS 2 topic data,
+* Triggering ROS 2 services or actions without being unauthorized,
+* Abusing node API to disrupt the robot behavior.
+
+However, vulnerabilities unrelated to ROS 2 are not in scope:
+
+* Checking that the robot SSH server cannot be accessed using “easy to guess” credentials such as user = root, password = admin for instance.
+* Validating that the robot NTP server or kernel is up-to-date and is not vulnerable to attacks.
+
+
+The rationale is that any arbitrary software package may be bundled with a particular robot. Trying to evaluate all possible vulnerabilities can be done using a generic vulnerability scanner or framework. We should not try to re-invent generic tools already developed by the security community.
+
+On the opposite, we need to address ROS 2 specific concerns as limited efforts have been done to document and mitigate those issues.
```

Comment:
Where do we draw the line e.g. what about a package depends on outdated library with known vulnerabilities? Or specific configuration issues?

Reply:
Generally, both of our examples would be out of the scope of this package, unless there is something ROS specific to it.

E.g. an openssh exploit is not in scope. But if the way `roslaunch` invokes `ssh` makes the exploit more dangerous, this is something that would be in scope.